spec/setup/helpers/abs_helper_spec.rb

```diff
@@ -0,0 +1,779 @@
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_INVALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_AWSDIRECT_RESPONSE_BODY)
+
+        allow(subject).to receive(:puts)
+        expect(subject).to receive(:puts).with(/Invalid ABS response/)
+        expect(subject).to receive(:puts).with(/code/)
+        expect(subject).to receive(:puts).with(/body/)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+    context 'when the response body is empty' do
+
+      it 'reports the error and returns false' do
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(TEST_INVALID_RESPONSE_BODY)
+
+        allow(subject).to receive(:puts)
+        expect(subject).to receive(:puts).with(/Invalid ABS response/)
+        expect(subject).to receive(:puts).with(/code/)
+        expect(subject).to receive(:puts).with(/body/)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+    context 'when the response body is nil' do
+
+      it 'reports the error and returns false' do
+        expect(test_http_response).to receive(:nil?).at_least(:once).and_return(false)
+        expect(test_http_response).to receive(:code).at_least(:once).and_return(TEST_VALID_RESPONSE_CODE)
+        expect(test_http_response).to receive(:body).at_least(:once).and_return(nil)
+
+        allow(subject).to receive(:puts)
+        expect(subject).to receive(:puts).with(/Invalid ABS response/)
+        expect(subject).to receive(:puts).with(/code/)
+        expect(subject).to receive(:puts).with(/body/)
+
+        expect(subject.abs_is_valid_response?(test_http_response)).to eq(false)
+      end
+
+    end
+
+  end
+
+  describe '#abs_get_a2a_hosts' do
+
+    before {
+      subject.instance_variable_set("@abs_aws_mom_size", TEST_ABS_AWS_MOM_SIZE)
+      subject.instance_variable_set("@abs_aws_metrics_size", TEST_ABS_AWS_METRICS_SIZE)
+    }
+
+    context 'when called' do
+
+      it 'initializes the helper and returns the a2a hosts' do
+        expect(subject).to receive(:abs_initialize)
+        expect(subject.abs_get_a2a_hosts).to eq(TEST_A2A_HOSTS)
+
+      end
+
+    end
+
+  end
+
+  describe '#abs_reformat_resource_host' do
+
+    context 'when a valid response_body is provided' do
+
+      it 'returns the expected reformatted response' do
+        expect(subject.abs_reformat_resource_host(TEST_AWSDIRECT_RESPONSE_BODY)).to eq(TEST_REFORMATTED_RESPONSE_BODY)
+      end
+
+    end
+
+    context 'when an invalid response_body is provided' do
```

Comment:
Looking at this test.  It doesn't feel like enough information is getting to the user to tell them what went wrong.  all they know is there was a json parse error somewhere...
And the method goes on to return nil.  The caller doesn't check it, just adds it to the list of responses, and appears to consider the nil the same way it does a valid conversion.

Reply:
I've updated this to provide more information.